plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsWriter.java

```diff
@@ -194,51 +189,58 @@ private CollectedStatistics mergeStatisticsIfNecessary(
+                .filter(blobMetadata -> blobMetadata.type().equals(StandardBlobTypes.APACHE_DATASKETCHES_THETA_V1))
+                .filter(blobMetadata -> blobMetadata.fields().size() == 1)
+                .anyMatch(blobMetadata -> columnsWithRecentlyComputedStats.contains(getOnlyElement(blobMetadata.fields())));
 
-                    if (hasUsefulData) {
-                        try (PuffinReader reader = Puffin.read(fileIO.newInputFile(statisticsFile.path()))
-                                .withFileSize(statisticsFile.fileSizeInBytes())
-                                .withFooterSize(statisticsFile.fileFooterSizeInBytes())
-                                .build()) {
-                            List<BlobMetadata> toRead = reader.fileMetadata().blobs().stream()
-                                    .filter(blobMetadata -> blobMetadata.type().equals(APACHE_DATASKETCHES_THETA_V1))
-                                    .filter(blobMetadata -> blobMetadata.inputFields().size() == 1)
-                                    .filter(blobMetadata -> pendingPreviousNdvSketches.contains(getOnlyElement(blobMetadata.inputFields())))
-                                    .collect(toImmutableList());
-                            for (Pair<BlobMetadata, ByteBuffer> read : reader.readAll(toRead)) {
-                                Integer fieldId = getOnlyElement(read.first().inputFields());
-                                checkState(pendingPreviousNdvSketches.remove(fieldId), "Unwanted read of stats for field %s", fieldId);
-                                Memory memory = Memory.wrap(ByteBuffers.getBytes(read.second())); // Memory.wrap(ByteBuffer) results in a different deserialized state
-                                CompactSketch previousSketch = CompactSketch.wrap(memory);
-                                CompactSketch newSketch = requireNonNull(collectedNdvSketches.get(fieldId), "ndvSketches.get(fieldId) is null");
-                                ndvSketches.put(fieldId, SetOperation.builder().buildUnion().union(previousSketch, newSketch));
-                            }
-                        }
-                        catch (IOException exception) {
-                            throw new UncheckedIOException(exception);
+        if (hasUsefulData) {
+            try (PuffinReader reader = Puffin.read(fileIO.newInputFile(statisticsFile.path()))
+                    .withFileSize(statisticsFile.fileSizeInBytes())
+                    .withFooterSize(statisticsFile.fileFooterSizeInBytes())
+                    .build()) {
+                List<BlobMetadata> toRead = reader.fileMetadata().blobs().stream()
+                        .filter(blobMetadata -> blobMetadata.type().equals(APACHE_DATASKETCHES_THETA_V1))
+                        .filter(blobMetadata -> blobMetadata.inputFields().size() == 1)
+                        .collect(toImmutableList());
+                for (Pair<BlobMetadata, ByteBuffer> read : reader.readAll(toRead)) {
```

Comment:
Why read anything in REPLACE mode?
i think we gonna not use the read results anyway?